r/R/dplyr-funcs-datetime.R

```diff
@@ -615,4 +618,44 @@ register_bindings_datetime_parsers <- function() {
 
     build_expr("assume_timezone", coalesce_output, options = list(timezone = tz))
   })
+
+}
+
+
+register_bindings_datetime_rounding <- function() {
+
+  register_binding("round_date", function(x, unit = "second",
+                                          week_start = getOption("lubridate.week.start", 7)) {
+    opts <- parse_period_unit(unit)
+    if (opts$unit == 7L) { # weeks (unit = 7L) are special
+      return(shift_temporal_to_week("round_temporal", x, week_start, options = opts))
+    }
+    Expression$create("round_temporal", x, options = opts)
+  })
+
+  register_binding("floor_date", function(x, unit = "second",
+                                          week_start = getOption("lubridate.week.start", 7)) {
+    opts <- parse_period_unit(unit)
+    if (opts$unit == 7L) { # weeks (unit = 7L) are special
+      return(shift_temporal_to_week("floor_temporal", x, week_start, options = opts))
+    }
+
+    return(Expression$create("floor_temporal", x, options = opts))
```

Comment:
We generally follow tidyverse style and only `return()` early: https://style.tidyverse.org/functions.html#return
```suggestion
    Expression$create("floor_temporal", x, options = opts)
```